flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/PreTransformOperator.java

```diff
@@ -411,18 +411,16 @@ private void cacheTransformRuleInfo(CreateTableEvent createTableEvent) {
 
     private CreateTableEvent transformCreateTableEvent(CreateTableEvent createTableEvent) {
         TableId tableId = createTableEvent.tableId();
-        for (Tuple2<Selectors, SchemaMetadataTransform> transform : schemaMetadataTransformers) {
-            Selectors selectors = transform.f0;
-            if (selectors.isMatch(tableId)) {
-                createTableEvent =
-                        new CreateTableEvent(
-                                tableId,
-                                transformSchemaMetaData(
-                                        createTableEvent.getSchema(), transform.f1));
-            }
-        }
-
-        cachePreTransformProcessor(tableId, createTableEvent.getSchema());
+        Schema originalSchema = createTableEvent.getSchema();
+        LinkedHashSet<Schema> newSchemas =
+                schemaMetadataTransformers.stream()
+                        .filter(transform -> transform.f0.isMatch(tableId))
+                        .map(transform -> transformSchemaMetaData(originalSchema, transform.f1))
+                        .collect(Collectors.toCollection(LinkedHashSet::new));
+        SchemaUtils.validateMetaSchemaCompatibility(newSchemas);
```

Comment:
We should throw an exception to clearly indicate which two rules conflict for which table.